edb/edgeql/compiler/stmtctx.py

```diff
@@ -272,6 +277,54 @@ def fini_expression(
     return result
 
 
+def _fts_lang_arg_fixup(
+    irs: Sequence[irast.Base], *, ctx: context.ContextLevel
+) -> None:
+    """Rewrite parameters that are cast to fts::language
+
+    FTS is very fiddly about its language arguments; it has an
+    internal type for them (regconfig), and it will not match indexes
+    if you produce it by casting from text, though casting directly
+    from a string literal works. This causes complications for us,
+    because our parameter extraction runs before anything.
+
+    We solve the problem by changing the type of any parameter that is
+    only used once and is directly cast to fts::language (likely
```

Comment:
Why the "only used once" restriction?  I think we can safely do this as long as every use is an `<fts::language><str>` cast.

Reply:
I did "only used once" because it was simpler to implement and because I was mostly thinking about the constant-extraction case, where each extracted constant is only used once, but I agree that we could do your thing and that it would be better in cases with an explicit <str> param.